test/e2e/runner.go

```diff
@@ -84,6 +88,7 @@ func main() {
 			Config:            engCfg,
 			ClusterDefinition: cs,
 		}
+		cliProvisioner.Engine = eng
```

Comment:
What is this doing? Does it assign it to null if the name of the cluster was provided?

Reply:
Ah nevermind I just realized this was in the else block

Reply:
It adds the ClusterDefinition (parsed api model) to the cliProvisioner instance, which is used to form ssh connection config to the master. And all of this convoluted stuff was done so that in cases where there is no provision (i.e., you're testing an existing cluster) we still have all the cluster metadata in that instance object during teardown (in this case not tearing down the cluster but grabbing logs).